lib/MusicBrainz/Server/Edit/Role/IPI.pm

```diff
@@ -12,6 +23,62 @@ with 'MusicBrainz::Server::Edit::Role::ValueSet' => {
     extract_value => sub { shift->ipi }
 };
 
+after initialize => sub {
+    my ($self, %opts) = @_;
+
+    my $old_ipis = $self->data->{old}{ipi_codes};
+    my $new_ipis = $self->data->{new}{ipi_codes};
+    my $added_ipis_set = Set::Scalar->new(@$new_ipis) - Set::Scalar->new(@$old_ipis);
+    my @added_ipis = $added_ipis_set->members;
+    $self->reused_ipis($self->c->model($self->_edit_model)->find_reused_ipis(@added_ipis));
+};
+
+after post_insert => sub {
+    my $self = shift;
+
+    if ($self->reused_ipis) {
+        for my $ipi (keys %{ $self->reused_ipis }) {
+            my $artist_dupe_count = $self->reused_ipis->{$ipi}->{artist};
+            my $label_dupe_count = $self->reused_ipis->{$ipi}->{label};
+            my $edit_note;
+            if ($artist_dupe_count && $label_dupe_count) {
+                 $edit_note = localized_note(N_l('The IPI {ipi} is already in use on both artists and labels ({artist_count} and {label_count} at the time of this note respectively). You can search for uses of this IPI on {artist_search|artists} and {label_search|labels}.'),
```

Comment:
Wouldn't it be exceptionally rare for both artists and labels to share an IPI or ISNI being added again? Not sure we need a specific message for this instead of just leaving two notes.

Reply:
Yes, it would, but I thought it would be more clear like this. I guess we could just leave two notes too.

Reply:
Dunno, it just seemed kinda weird to have a string for a situation that might happen once a year, but I could be underestimating people. :)